runtime/test/test_argument_conversion.cpp

```diff
@@ -15,40 +15,67 @@
 #include "common/ArgumentConversion.h"
 #include "cudaq/Optimizer/Dialect/CC/CCDialect.h"
 #include "cudaq/Optimizer/Dialect/Quake/QuakeDialect.h"
+#include "cudaq/qis/pauli_word.h"
 #include "mlir/InitAllDialects.h"
 #include "mlir/Parser/Parser.h"
+#include <sstream>
 
-void doSimpleTest(mlir::MLIRContext *ctx, const std::string &typeName,
-                  std::vector<void *> args) {
-  std::string code = R"#(
-func.func private @callee(%0: )#" +
-                     typeName + R"#()
-func.func @__nvqpp__mlirgen__testy(%0: )#" +
-                     typeName + R"#() {
-  call @callee(%0) : ()#" +
-                     typeName + R"#() -> ()
-  return
-})#";
+void doTest(mlir::MLIRContext *ctx, std::vector<std::string> &typeNames,
+            std::vector<void *> args, std::size_t startingArgIdx = 0) {
+
+  // Create code
+  std::ostringstream os;
+  os << "func.func private @callee(%0: " << typeNames[0];
+  for (auto i = 1; i < typeNames.size(); i++)
+    os << ", %" << i << ": " << typeNames[i];
+  os << ")\n";
+
+  os << "func.func @__nvqpp__mlirgen__testy(%0: " << typeNames[0];
+  for (auto i = 1; i < typeNames.size(); i++)
+    os << ", %" << i << ": " << typeNames[i];
+  os << ") {\n";
+
+  os << "  call @callee(%0";
+  for (auto i = 1; i < typeNames.size(); i++)
+    os << ", %" << i;
+  os << "): ";
+
+  os << "(" << typeNames[0];
+  for (auto i = 1; i < typeNames.size(); i++) {
+    os << ", " << typeNames[i];
+  }
+  os << ") -> ()\n";
+
+  os << "  return\n";
+  os << "}\n";
+
+  auto code = os.str();
 
   // Create the Module
   auto mod = mlir::parseSourceString<mlir::ModuleOp>(code, ctx);
   llvm::outs() << "Source module:\n" << *mod << '\n';
   cudaq::opt::ArgumentConverter ab{"testy", *mod};
 
   // Create the argument conversions
-  ab.gen(args);
+  ab.gen_drop_front(args, startingArgIdx);
```

Comment:
Why are we removing the original test that uses `gen` here?

Reply:
The `gen_drop_front` wasn't tested and had an issue, so I was making sure were are going through all the paths (`gen_drop_front` calls `gen` eventually, but if you would prefer I can create separate tests for `gen_drop_front` that would look the same except calling `gen_drop_front` instead of `gen`.

Reply:
I restored the original tests and only now use `gen_drop_front` in an additional `test_combinations` set of tests, to make sure the issues are fixed.